TeachingRecordSystem/src/TeachingRecordSystem.Core/Dqt/DataverseAdapter.UpdateTeacher.cs

```diff
@@ -226,16 +226,23 @@ public async Task<UpdateTeacherResult> UpdateTeacher(UpdateTeacherCommand comman
             });
         }
 
-        if (referenceData.TeacherHusId != command.HusId || referenceData.Teacher.dfeta_SlugId != command.SlugId)
+        if (referenceData.TeacherHusId != command.HusId || referenceData.Teacher.dfeta_SlugId != command.SlugId || (referenceData.HavePendingPiiChanges == false && helper.ShouldUpdatePII(referenceData.Teacher.dfeta_AllowPiiUpdatesFromRegister) == true))
```

Comment:
I think these conditions need to be separated; presumably we wouldn't want `dfeta_HUSID` to potentially be nulled out because we're updating PII.

Reply:
I've left this as is because as i've changed the types to be option<string> so husid/slugid will only update if there is a value.